packages/wrangler/src/index.tsx

```diff
@@ -1335,6 +1335,7 @@ export async function main(argv: string[]): Promise<void> {
             try {
               await submitSecret();
             } catch (e) {
+              // @ts-expect-error non-standard property on Error
               if (e.code === 10007) {
```

Comment:
Maybe we should extend Error instead of just adding the `code` on to it.

Reply:
Definitely up for creating a `FetchError`, where the logic from `throwFetchError()` gets put into its constructor.

Reply:
Yeah, we need to redo errors, further to aid in reporting. I'll do this in a followup PR with @JacobMGEvans